### PR TITLE
feat!: add to test coverage, includes some new funcs and breaking changes

### PR DESCRIPTION
**Pull request by amilumibe**

### CODE REVIEW

File: coins/bitcoin/src/bitcoin_wallet.rs [hunk 1]
```diff
@@ -348,6 +353,39 @@ impl BitcoinWallet {
         }
     }
 
+    /// Adds an address on the first account (account_index = 0) with the specified address index to the [BitcoinWallet]
+    /// If the particular address is already associated with the wallet, it will be added again
+    ///
+    /// If not enough info is kn
```
Comment (devinpearson) [inline coins/bitcoin/src/bitcoin_wallet.rs:359]: Not sure this sentence is finished

File: coins/bitcoin/src/bitcoin_wallet.rs [hunk 2]
```diff
@@ -422,10 +469,15 @@ impl BitcoinWallet {
             .address_index(max_address + 1)
             .build()
             .to_string();
-        let next_hd_key = self.master_hd_key()?.derive(next_deriv_path)?;
+        let next_hd_key = self.master_hd_key()?.derive(&next_deriv_path)?;
         BitcoinAddress::from_hd_key(&next_hd_key, self.address_format)
     }
 
+    /// Sets the [master hd key][HDKey] associated with the [wallet][BitcoinWallet].
```
Comment (devinpearson) [inline coins/bitcoin/src/bitcoin_wallet.rs:476]: What's the need for this setter?

Comment (amilumibe) [reply coins/bitcoin/src/bitcoin_wallet.rs:476]: adding this function allows the master key to be set directly without going through the builder, without this function, only way to set this field is by using  the builder pattern

File: coins/bitcoin/src/bitcoin_wallet.rs (hunk 1 above)
Comment (amilumibe) [reply coins/bitcoin/src/bitcoin_wallet.rs]: I've now finished this sentence

File: coins/bitcoin/src/bitcoin_wallet.rs (hunk 2 above)
Comment (devinpearson) [reply coins/bitcoin/src/bitcoin_wallet.rs:476]: ok no problem, Ill want to review this later

**Review by devinpearson — APPROVED** [review]

**Pull request merged**